tests/Doctrine/Tests/ORM/Tools/DebugTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Tools;
+
+use ArrayIterator;
+use ArrayObject;
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+use Doctrine\ORM\Tools\Debug;
+use Doctrine\Tests\DoctrineTestCase;
+use stdClass;
+
+use function print_r;
+use function strpos;
+use function substr;
+
+class DebugTest extends DoctrineTestCase
+{
+    public function testExportObject(): void
+    {
+        $obj      = new stdClass();
+        $obj->foo = 'bar';
+        $obj->bar = 1234;
+
+        $var = Debug::export($obj, 2);
+        self::assertEquals('stdClass', $var->__CLASS__);
+    }
+
+    public function testExportObjectWithReference(): void
+    {
+        $foo = 'bar';
+        $bar = ['foo' => & $foo];
+        $baz = (object) $bar;
+
+        $var      = Debug::export($baz, 2);
+        $baz->foo = 'tab';
+
+        self::assertEquals('bar', $var->foo);
+        self::assertEquals('tab', $bar['foo']);
+    }
+
+    public function testExportArray(): void
+    {
+        $array              = ['a' => 'b', 'b' => ['c', 'd' => ['e', 'f']]];
+        $var                = Debug::export($array, 2);
+        $expected           = $array;
+        $expected['b']['d'] = 'Array(2)';
+        self::assertEquals($expected, $var);
+    }
+
+    public function testExportDateTime(): void
+    {
+        $obj = new DateTime('2010-10-10 10:10:10', new DateTimeZone('UTC'));
+
+        $var = Debug::export($obj, 2);
+        self::assertEquals('DateTime', $var->__CLASS__);
+        self::assertEquals('2010-10-10T10:10:10+00:00', $var->date);
+    }
+
+    public function testExportDateTimeImmutable(): void
+    {
+        $obj = new DateTimeImmutable('2010-10-10 10:10:10', new DateTimeZone('UTC'));
+
+        $var = Debug::export($obj, 2);
+        self::assertEquals('DateTimeImmutable', $var->__CLASS__);
+        self::assertEquals('2010-10-10T10:10:10+00:00', $var->date);
+    }
+
+    public function testExportDateTimeZone(): void
+    {
+        $obj = new DateTimeImmutable('2010-10-10 12:34:56', new DateTimeZone('Europe/Rome'));
+
+        $var = Debug::export($obj, 2);
+        self::assertEquals('DateTimeImmutable', $var->__CLASS__);
+        self::assertEquals('2010-10-10T12:34:56+02:00', $var->date);
+    }
+
+    public function testExportArrayTraversable(): void
+    {
+        $obj = new ArrayObject(['foobar']);
+
+        $var = Debug::export($obj, 2);
+        self::assertContains('foobar', $var->__STORAGE__);
+
+        $it = new ArrayIterator(['foobar']);
+
+        $var = Debug::export($it, 5);
+        self::assertContains('foobar', $var->__STORAGE__);
+    }
+
+    /**
+     * @param array<string, int> $expected
+     *
+     * @dataProvider provideAttributesCases
+     *
+     * @requires PHP < 8.1.0
```

Comment:
Why?

Reply:
Apparently, the representation differs. I can make the data provider output different orders based on the PHP version so that we continue testing this on 3.0

Reply:
Do the representations only differ in the order of the array keys? I think PHP 8 changed the order in which parent and child properties are dumped.

If that's the case, let's put the PHP 8.1 representation into the data provider and if we're on PHP 7, we `ksort()` both arrays before comparing them.

Reply:
Yes, that's the only difference. I'll try what you are recommending.

Reply:
`ksort` does not do the trick the order seems to be alphabetical, then public, protected, private :P

Reply:
Alternatively, if we don't care, I could use https://docs.phpunit.de/en/10.4/assertions.html#assertequalscanonicalizing

Reply:
Yes, let's try that.

Reply:
Actually that won't work: we're comparing strings (generated by `print_r`) here, not arrays. Besides, on 3.0.x, this canonicalization will be useless. As it stands, the code makes it more clear that something can be removed.

Reply:
All right.